### PR TITLE
[DPE-3112] Fix DEPENDENCIES field mismatch

### PR DESCRIPTION
**Pull request by deusebio**

### CODE REVIEW

**Review by marcoppenheimer — APPROVED** [review]

File: src/charm.py [hunk 1]
```diff
@@ -392,6 +393,8 @@ def _disable_enable_restart(self, event: RunWithLock) -> None:
         self.snap.disable_enable()
         self.snap.start_snap_service()
 
+        time.sleep(20.0)
+
```
Comment (zmraul) [inline src/charm.py:397]: question: Is this specifically related to storage being added?

Comment (deusebio) [reply src/charm.py:397]: No, no, this is in Kafka. When testing the upgrades under load I noticed that rolling restart units too quick (without timeouts) could get the clients a bit in trouble. Basically, a consumer would try to refresh the metadata and use a new broker to fetch data, but that broken may have meanwhile gone offline, screwing a bit the process. tens of seconds seemed good enough to get things a bit more stable and robust to allow for broker rejoining, catching up and also clients transitioning from one broker to another

Comment (marcoppenheimer) [reply src/charm.py:397]: I think what Raul meant here, is that `disable_enable_restart` is a hack call we do to get around snaps not loading the new storages until it's 're-enabled'. I'm not sure that's needed for upgrades?

File: src/upgrade.py [hunk 2]
```diff
@@ -101,6 +102,9 @@ def _on_upgrade_granted(self, event: UpgradeGrantedEvent) -> None:
         logger.info(f"{self.charm.unit.name} upgrading service...")
         self.charm.snap.restart_snap_service()
 
+        # Allow for some time to settle down
+        time.sleep(20)
+
```
Comment (marcoppenheimer) [inline src/upgrade.py:107]: **todo:** I would mark this as `FIXME`, but it's probably fine for the short term.

I don't think we want to rely on a hard-coded value (any value really). We can use the `pause` upgrade-strategy that was mentioned recently in the newer upgrade specs for production deployments, which avoids this issue.

Or, we need to add more detailed 'ready' checks, and use them.

**Review by marcoppenheimer — APPROVED** [review]

File: src/upgrade.py (hunk 2 above)
Comment (deusebio) [reply src/upgrade.py:107]: Posting also here: https://warthogs.atlassian.net/browse/DPE-3155

**Review by zmraul — APPROVED** [review]
